pkg/reconciler/pipelinerun/resources/pipelineref.go

```diff
@@ -17,13 +17,62 @@ limitations under the License.
 package resources
 
 import (
+	"context"
 	"fmt"
 
+	"github.com/google/go-containerregistry/pkg/authn/k8schain"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
 	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
 	clientset "github.com/tektoncd/pipeline/pkg/client/clientset/versioned"
+	"github.com/tektoncd/pipeline/pkg/remote/oci"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
 )
 
+// GetPipelineFunc is a factory function that will use the given PipelineRef to return a valid GetPipeline function that
+// looks up the pipeline. It uses as context a k8s client, tekton client, namespace, and service account name to return
+// the pipeline. It knows whether it needs to look in the cluster or in a remote image to fetch the reference.
+func GetPipelineFunc(k8s kubernetes.Interface, tekton clientset.Interface, pr *v1beta1.PipelineRef, namespace, saName string) (GetPipeline, error) {
+	switch {
+	case pr != nil && pr.Bundle != "":
+		// If there is a bundle url at all, construct an OCI resolver to fetch the pipeline.
+		kc, err := k8schain.New(k8s, k8schain.Options{
+			Namespace:          namespace,
+			ServiceAccountName: saName,
+		})
+		if err != nil {
+			return nil, fmt.Errorf("failed to get keychain: %w", err)
+		}
+		resolver := oci.NewResolver(pr.Bundle, kc)
+		// Return an inline function that implements GetTask by calling Resolver.Get with the specified task type and
+		// casting it to a TaskInterface.
+		return func(name string) (v1beta1.PipelineInterface, error) {
```

Comment:
Any reason for using an inline function here instead of extracting it ?

Reply:
I figured it was short enough that it would be more useful to have the definition near the usage.